Promxy panics during startup when web.Options.ListenAddresses is empty.

## Describe the bug

Promxy panics during startup when `web.Options.ListenAddresses` is empty.

The panic occurs in:

```go id="6l80fx"
// web/web.go:367
api_v1.GlobalURLOptions{
    ListenAddress: o.ListenAddresses[0],
    Host:          o.ExternalURL.Host,
    Scheme:        o.ExternalURL.Scheme,
}
```

If `ListenAddresses` is not initialized, this causes:

```text id="72vvt9"
panic: runtime error: index out of range [0] with length 0
```

## Root cause

In `cmd/promxy/main.go`, `web.Options` is created without setting `ListenAddresses`:

```go id="bx0e2g"
webOptions := &web.Options{
    // ... other fields
    // ListenAddresses not set
}
```

Then:

```go id="8ffr59"
webHandler := web.New(logger, webOptions)
```

This eventually dereferences `o.ListenAddresses[0]` without checking slice length.

## Expected behavior

Promxy should avoid panicking when `ListenAddresses` is empty.

Possible approaches:

* validate the slice before indexing
* provide a default listen address
* return a descriptive startup error instead of panicking

## Possible fix

```go id="5r6k0t"
listenAddress := ""
if len(o.ListenAddresses) > 0 {
    listenAddress = o.ListenAddresses[0]
}
```

## Environment

* promxy version: <version>
* Go version: <version>
* OS: <os>

## Additional context

This appears to happen because Promxy does not initialize `ListenAddresses` before calling `web.New()`.

--- a/cmd/promxy/main.go
+++ b/cmd/promxy/main.go
@@ -421,6 +421,12 @@ func main() {
 
 		EnableLifecycle: opts.EnableLifecycle,
 
+		// Prometheus' web.New() indexes ListenAddresses[0] unconditionally when
+		// constructing GlobalURLOptions; promxy doesn't use the embedded
+		// listeners (it has its own server), but the slice still must be set
+		// or startup panics.
+		ListenAddresses: []string{opts.BindAddr},
+
 		Flags:       opts.ToFlags(),
 		RoutePrefix: opts.RoutePrefix,
 		ExternalURL: externalUrl,

